### PR TITLE
Refactor model & response data

### PR DESCRIPTION
**Pull request by sleepylee**
## What happened 👀
Before we move to integrate with the GraphQL API client, we should clean up the HTTP API client base.
 
## Insight 📝
- Moving packages & doing a small merging, the data returned from the raw response from an HTTP API client, should be called explicitly as `XxResponse` - then an app actual `model` class should be called as pure `Xx`. In this case we refactor to have: `AuthTokenResponse` and `AuthToken` separately. This creates better separation of concerns between multiple sources of data (upcoming graphQL), with the actual app's data. They can live together when it becomes app data, but before that, the raw data should come from distinct sources. 
 
## Proof Of Work 📹

The app works normally. Some dependencies were added ahead for the graphql integration shouldn't be the concern.

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

File: lib/converters/auth_token_converter.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'package:survey/api/http/response/auth_token_response.dart';
+import 'package:survey/models/auth_token.dart';
+
+extension AuthTokenConverter on AuthTokenResponse {
```
Comment (manh-t) [inline lib/converters/auth_token_converter.dart:4]: I think we can reduce 1 class by adding this `extension` function into `auth_token.dart` like our DM project 😄

File: lib/use_cases/get_login_state_use_case.dart [hunk 2]
```diff
@@ -8,8 +8,9 @@ class GetLoginStateUseCase extends NoParamsUseCase<bool> {
 
   @override
   Future<Result<bool>> call() {
-    return _sharedPreferencesStorage
-        .getAccessToken()
-        .then((token) => Success(token != null && token.isNotEmpty));
+    return _sharedPreferencesStorage.getAccessToken().then((token) {
+      print(token);
+      return Success(token != null && token.isNotEmpty);
+    });
```
Comment (manh-t) [inline lib/use_cases/get_login_state_use_case.dart:14]: redundant `print`? 💇

**Comment by sleepylee** [comment]
all fixed, @manh-t 🙇

**Review by manh-t — APPROVED** [review]
LGTM 🚀

**Pull request merged**